### DISCUSSION TITLE
NPT simulations with LAMMPS+Allegro

### DISCUSSION THREAD

**Discussion by EnricoTrizio** [mir-group/allegro#4] [Q&A] [answered]
Morning guys,

I've used NequIP before and now I'm testing Allegro for the study of some phase transition processes in the future, thus I need to run NPT simulations on my systems. 
The patch with LAMMPS is super convenient as it can run very fast but it's not clear to me if the pair_allegro patch can be safely used with barostats to run NPT. So far, I tried kind of blindly and always got strange results for systems in which NVT was fine and stable, so maybe I've been missing something even if it doesn't raise any error.
I've also seen that for NequIP you have a stress/develop branch on Github even if only compatible with ASE and still under testing, so I was wondering about the possibilities with Allegro.

In case the answer is no, is this something you are planning to add in the near future and/or can you give me eventually some indications about where these things should be handled within the code?

Thanks in advance for the help!

**Comment by simonbatzner** [comment, marked as answer]
Answered in email, summary: stress fully working on the PyTorch side, see `develop` branch of nequi repo. Stress in LAMMPS is implemented in `pair_nequip` (for NequIP and Allegro, albeit Allegro less well-tested). This let's you do single-GPU NPT runs. We're working on a parallel `pair_allegro` implementation of stress.

**Reply by sigbjobo** [reply]
Hi, what is the status of the NPT implementation?

**Reply by Linux-cpp-lisp** [reply]
Hi @sigbjobo 

Everything should now be working on the `stress` branch of `pair_allegro` combined with the latest `develop` of `nequip`.  That said, it's still pretty new so please let us know if you encounter any problems.

**Reply by Linux-cpp-lisp** [reply]
See also https://github.com/mir-group/pair_allegro/issues/12

**Comment by weilaigit** [comment]
To follow up on this, in order to use stress/LAMMPS/Allegro, we need to use the develop branch of nequip and the stress branch of pair_allegro.   What about allegro, main or develop branch?  Thanks.
